### PR TITLE
Image query request protection

### PR DESCRIPTION
**Pull request by whyleee**
Image query request protection to prevent DOS attacks from third-party image clients.

### CODE REVIEW

File: RestImageResize/Security/QueryAuthorizer.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace RestImageResize.Security
+{
+    /// <summary>
+    /// Authorizes image transformation queries.
+    /// </summary>
+    public class QueryAuthorizer
```
Comment (VladimirLevchuk) [inline RestImageResize/Security/QueryAuthorizer.cs]: would be really nice to create IQueryAuthorized contract so user can override authorization using DI

File: RestImageResize/OpenWaveRestApiEncoder.cs [hunk 2]
```diff
@@ -110,5 +116,17 @@ public Uri EncodeUri(Uri uri)
...
         }
+
+        private void EnsureAuthorizedQuery(ImageTransformQuery query)
+        {
+            var authorized = QueryAuthorizer.IsAuthorized(query);
+
+            if (!authorized)
+            {
+                throw new HttpException((int) HttpStatusCode.Forbidden,
+                    string.Format("Forbidden query 'w={0}&h={1}&t={2}&h={3}'",
```
Comment (VladimirLevchuk) [inline RestImageResize/OpenWaveRestApiEncoder.cs]: I'd remove parameters from error message

File: RestImageResize/web.config.transform [hunk 3]
```diff
@@ -2,6 +2,7 @@
 <configuration>
   <appSettings>
     <add key="RestImageResize.DefautTransform" value="downFit"/>
+    <add key="RestImageResize.PrivateKeys" value="" />
```
Comment (VladimirLevchuk) [inline RestImageResize/web.config.transform]: maybe it makes sense to add a comment like "put semicolon separated private keys" here to start securing your image requests"

File: RestImageResize/OpenWaveRestApiEncoder.cs (hunk 2 above)
Comment (whyleee) [inline RestImageResize/OpenWaveRestApiEncoder.cs]: They help to debug. And we're not exposing anything internal: these are parameters from request.

File: RestImageResize/web.config.transform (hunk 3 above)
Comment (whyleee) [inline RestImageResize/web.config.transform]: Then all the settings should be commented ;) Now we have everything in readme - so this should be another pull request.
